central/postgres/pruning.go

```diff
@@ -47,6 +48,42 @@ const (
 		(SELECT 1 FROM pods WHERE pi.poduid = pods.Id)) 
 		delete FROM process_indicators pi USING orphan_proc op WHERE pi.id = op.id AND 	
 		(signal_time < now() AT time zone 'utc' - INTERVAL '%d MINUTES' OR signal_time IS NULL)`
+
+	// (snapshots.reportstatus_runstate = 3 OR snapshots.reportstatus_runstate = 5)
+	// ...gives us the report jobs that are in final state.
+	//
+	// (SELECT MAX(latest.reportstatus_completedat) FROM ` + schema.ReportSnapshotsTableName + ` latest
+	// WHERE latest.reportstatus_completedat IS NOT NULL
+	// AND snapshots.reportconfigurationid = latest.reportconfigurationid
+	// AND latest.reportstatus_runstate = 5
+	// GROUP BY latest.reportstatus_reportnotificationmethod, latest.reportstatus_reportrequesttype)
+	//	...gives us the last successful report job for each config, for each notificated method, and each request type.
+	//
+	// (SELECT 1 FROM ` + schema.BlobsTableName + ` blobs
+	//	WHERE blobs.name not ilike '%/snapshots.reportid')
+	// ...tells us if the report still exists in the blob store.
+	//
+	// (reportstatus_completedat < now() AT time zone 'utc' - INTERVAL '%d MINUTES')
+	// ...gives us the reports that are outside the retention window.
+	pruneOldReportHistory = `DELETE FROM ` + schema.ReportSnapshotsTableName + ` WHERE reportid IN
```

Comment:
1. The comments are fantastic.  Extremely helpful

2. What does the explain plan look like for this query?  I'm wondering if query would be more efficient if we took the `SELECT MAX...` portion and pulled it into a `WITH` clause.  But that mostly depends on what the explain plan of this one looks like along with projected amount of records we are dealing with.  Basically I'm not sure if we care to fully optimize or not.

IMO at a minimum we should get an explain plan on this and if it looks like it may be problematic down the road, add a ticket to https://issues.redhat.com/browse/ROX-18983.